### PR TITLE
SEO Tools: add archive_title page title token

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request

Related (prerequisite): https://github.com/Automattic/jetpack/pull/20920

This PR:

* Supports adding & previewing a more generic `archive_title` token for custom Archive page titles.
* Hides the `Date` token button for Archives (but we still tokenize existing [date] usage).
* The `Date` button will only be replaced with the `Archive Title` button on Jetpack versions `>= 10.2-alpha`

#### Testing instructions

* See: https://github.com/Automattic/jetpack/pull/20920

---

Screenshot for Translators:

Markup on 2021-09-10 at 13:33:55

### CODE REVIEW

File: client/components/seo/meta-title-editor/index.jsx [hunk 1]
```diff
@@ -23,14 +23,15 @@ const getValidTokens = ( translate ) => ( {
 	pageTitle: translate( 'Page Title' ),
 	groupTitle: translate( 'Tag or Category Name' ),
 	date: translate( 'Date' ),
+	archiveTitle: translate( 'Archive Title' ),
```
Comment (a8ci18n) [inline client/components/seo/meta-title-editor/index.jsx:26]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Testimonial Archive Title' )` *ES Score: 6*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/seo-archive-title)

**Comment by samiff** [comment]
@jeherve Does the version check added in https://github.com/Automattic/wp-calypso/pull/56084/commits/e4e21faeb74fd2484b0aca4c06f23167fdad92dc look ok?

**Review by jeherve — COMMENTED** [review]
> Does the version check added in e4e21fa look ok?

Yup. Although you could use the `isJetpackMinimumVersion` utility, to have less work to do?

https://github.com/Automattic/wp-calypso/blob/7655b6d1f326371990851da10ebb97f133fbdc17/client/state/sites/selectors/is-jetpack-minimum-version.js#L5-L15

**Comment by samiff** [comment]
> Although you could use the `isJetpackMinimumVersion` utility, to have less work to do?

Thanks! Totally missed that one.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6581579

Thank you @samiff for including a screenshot in the description! This is really helpful for our translators.

**Comment by samiff** [comment]
@jeherve Small snag here, `isJetpackMinimumVersion` or fetching `getSiteOption( state, siteId, 'jetpack_version' )` won't return truthy for a WPcom non-AT Business plan site which should have the SEO settings.

Would this work, or is there a better way?

```js
	let shouldShowSeoArchiveTitleButton = false;
	if ( isJetpackMinimumVersion( state, siteId, '10.2-alpha' ) ) {
		shouldShowSeoArchiveTitleButton = true;
	} else if (
		! isJetpackSite( state, siteId ) &&
		hasActiveSiteFeature( state, siteId, FEATURE_ADVANCED_SEO )
	) {
		// For non-AT Business plan sites which get SEO features.
		shouldShowSeoArchiveTitleButton = true;
	}
```

**Comment by jeherve** [comment]
Yeah, that sounds good to me.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6581579).

**Pull request merged**